src/radical/pilot/utils/prof_utils.py

```diff
@@ -23,182 +24,484 @@
+                         {ru.EVENT: 'app_stop'               }],
+        'term_app'    : [{ru.EVENT: 'app_stop'               },
+                         {ru.EVENT: 'prte_iof_complete'      }],
+        'prte_phase_3': [{ru.EVENT: 'prte_iof_complete'      },
+                         {ru.EVENT: 'prte_notify_completed'  }],
+        'prte_phase_4': [{ru.EVENT: 'prte_notify_completed'  },
+                         {ru.EVENT: 'cu_exec_stop'           }],
+        'term_sh'     : [{ru.EVENT: 'cu_exec_stop'           },
+                         {ru.EVENT: 'cu_stop'                }],
+        'term_rp'     : [{ru.EVENT: 'cu_stop'                },
+                         {ru.EVENT: 'exec_stop'              }],
+        'unschedule'  : [{ru.EVENT: 'exec_stop'              },
+                         {ru.EVENT: 'unschedule_stop'        }]
+    }
 }
 
+
+# ----------------------------------------------------------------------------
+#
+def _convert_sdurations(sdurations):
+    '''
+    Converts a collection of durations expressed in short form to the same
+    collection of durations expressed in long form.
+
+    Definitions:
+
+    - Short form collection: one dictionary of short form durations
+    - Long form: one dictionary of long form durations.
+
+    Args:
+
+        sdurations (dict): a collections of durations in short form
+
+    Return:
+
+        ldurations (dict): a collection of long form durations
+
+    Example:
+
+        sdurations = {'name_of_duration': [{'STATE': s.STATE_NAME},
+                                           {'EVENT': 'event_name'}]}
+        ldurations = {'name_of_duration': [{ru.EVENT: 'state',
+                                            ru.STATE: s.STATE_NAME},
+                                           {ru.EVENT: 'event_name',
+                                            ru.STATE: None}]}
+        sdurations = {'name_of_duration': [{'STATE': s.STATE_NAME},
+                                           [{'EVENT': 'event_name'},
+                                            {'STATE': s.STATE_NAME}]]}
+        ldurations = {'name_of_duration': [{ru.EVENT: 'state',
+                                            ru.STATE: s.STATE_NAME},
+                                            [{ru.EVENT: 'event_name',
+                                              ru.STATE: None},
+                                             {ru.EVENT: 'state',
+                                              ru.STATE: s.event_name}]}
+        sdurations = {'name_of_duration': [{'STATE': s.STATE_NAME},
+                                           {'MSG': 'message_name'}]}
+        ldurations = {'name_of_duration': [{ru.EVENT: 'state',
+                                            ru.STATE: s.STATE_NAME},
+                                           {ru.EVENT: 'cmd',
+                                            ru.MSG: 'message_name'}]}
+    '''
+
+    ldurations = {}
+
+    for k,v in sdurations.items():
+
+        ldurations[k] = []
```

Comment:
You may want to use 
```
ldurations[k] = list()
```
more explicit, and easier to read to people not yet fluent in Python.  I mean, `[]` for list is pretty clear to most, but it is getting confusing when looking at tuples and sets for instance, so we elsewhere usually use the class names for clarity.

Reply:
Done.